runtime/lua/vim/lsp/util.lua

```diff
@@ -686,6 +686,20 @@ do
     severity_highlights[severity] = highlight_name
   end
 
+  function M.buf_get_diagnostics(bufnr)
+    validate { bufnr = {bufnr, 'n', true} }
+    bufnr = bufnr == 0 and api.nvim_get_current_buf() or bufnr
```

Comment:
If `bufnr` is `nil` or `vim.NIL`, this function raise error at line:694.
Why is validation optional true?